(UI) Modular Theme

Deadlineem opened the pull request:
- Added Modular Theme
- Moved SetupStyle function to Themes.cpp for future impl
- Ensured all themes use proper colors using ImGui::GetColorU32 to pull colors used in SetupStyle()
- Added more ImGui items to the SetupStyle function for a broader range of colors, allowing users to create their variations of themes in more than just the "Classic" style way

Adding the SetupStyle to Themes.cpp allows us to alter this easier in the future for adding Color selection options to the menu, giving users the ability to customize the menu colors to their personal preferences rather than just sticking to one single built in color.

// TODO:  Add ImGui Color pickers and allow setting custom values for use with SetupStyle

![Screenshot_148](https://github.com/user-attachments/assets/428217b1-9263-4ab5-ad02-243eaaf2466b)
![Screenshot_150](https://github.com/user-attachments/assets/8d416220-0aef-429c-bbc9-9632e3e95fb7)
![Screenshot_151](https://github.com/user-attachments/assets/5a869fe6-4fd6-400c-bd9f-d7e35f1c99b9)
![Screenshot_152](https://github.com/user-attachments/assets/e550e3a8-0dbc-4f60-b16c-88a78169a0be)
![Screenshot_156](https://github.com/user-attachments/assets/67b11ee5-e9a6-4af9-aca4-c29858f7d59c)

maybegreat48 commented on src/core/frontend/manager/UIManager.cpp:
```diff
@@ -9,6 +9,7 @@ namespace YimMenu::Features
 		{0, "Classic"},
 		{1, "Modern"},
 		{2, "Modern (Vertical)"},
+	    {3, "Modular"},
```
Maybe "Modern (Modular)"?

Deadlineem replied:
> Maybe "Modern (Modular)"?

Possibly, I was looking for feedback as well on if i should alter this style differently from the modern theme (like either sharper edges or making the buttons round?).  I was also considering a YimMenuV1 theme, I know a lot of people have been asking me about that, what do you think?

Deadlineem commented:
![Screenshot_161](https://github.com/user-attachments/assets/33e2a0fc-1e77-4ad3-87fd-9803a18f72f5)
![Screenshot_162](https://github.com/user-attachments/assets/90cfa24a-1e38-428d-81a1-c34d14b8f97f)

ImGui Color Settings/Rounding Settings for customization

### THIS THROWS EXCEPTIONS

**Would appreciate some assistance in fixing the exceptions this throws, otherwise it works nicely.**

Deadlineem commented:
Update:  Fixed some stuff, added more stuff.  Pretty much done aside from that exception it throws.

![Screenshot_163](https://github.com/user-attachments/assets/9198b1b0-cd50-4762-b49c-3e77541298f7)
![Screenshot_164](https://github.com/user-attachments/assets/20dcb375-7512-4a6f-89a9-344d35de6a5c)
![Screenshot_165](https://github.com/user-attachments/assets/0b76fa77-7aac-49da-8e8f-c3502992ae28)
![Screenshot_166](https://github.com/user-attachments/assets/75b94de5-f6a2-460c-9575-789dd6808950)
![Screenshot_167](https://github.com/user-attachments/assets/d08af239-8416-4531-b6ce-b69d137774d2)

Mr-X-GTA commented on src/game/frontend/submenus/Settings/GUISettings.cpp:
```diff
@@ -0,0 +1,348 @@
+	{
+		InitializeColorCommands();
+
+		ImGui::Text("ImGui Style Editor");
+		ImGui::Separator();
+
+		if (ImGui::BeginTabBar("StyleTabs"))
+		{
+			if (ImGui::BeginTabItem("Colors"))
+			{
+				DrawColorsTab();
+				ImGui::EndTabItem();
+			}
+			if (ImGui::BeginTabItem("Rounding"))
+			{
+				DrawRoundingTab();
+				ImGui::EndTabItem();
+			}
+			if (ImGui::BeginTabItem("Layout"))
+			{
+				DrawLayoutTab();
+				ImGui::EndTabItem();
+			}
+			if (ImGui::BeginTabItem("Alignment"))
+			{
+				DrawAlignmentTab();
+				ImGui::EndTabItem();
+			}
+			if (ImGui::BeginTabItem("Fonts"))
+			{
+				DrawFontTab();
+				ImGui::EndTabItem();
+			}
+			ImGui::EndTabBar();
+		}
```
Move this to ImGuiEditor->AddItem

```
ImGuiEditor->AddItem(std::make_unique<ImGuiItem>([] {
	ImGui::Text("ImGui Style Editor");
	...
}));
```

Deadlineem replied:
> Move this to ImGuiEditor->AddItem
> 
> ```
> ImGuiEditor->AddItem(std::make_unique<ImGuiItem>([] {
> 	ImGui::Text("ImGui Style Editor");
> 	...
> }));
> ```

Thannk You!
